massa-async-pool/src/pool.rs

```diff
@@ -52,13 +59,16 @@ impl AsyncPool {
         for change in changes.0.iter() {
             match change {
                 // add a new message to the pool
-                Change::Add(msg_id, msg) => {
-                    self.messages.insert(*msg_id, msg.clone());
+                Change::Add(message_id, message) => {
+                    self.messages.insert(*message_id, message.clone());
+                    self.hash ^= message.hash;
                 }
 
                 // delete a message from the pool
-                Change::Delete(msg_id) => {
-                    self.messages.remove(msg_id);
+                Change::Delete(message_id) => {
+                    if let Some(removed_message) = self.messages.remove(message_id) {
+                        self.hash ^= removed_message.hash;
+                    }
                 }
             }
         }
```

Comment:
In the function `settle_slot` just after there is `self.messages` that is extended and popped and there is no modifications of the hash.

Reply:
settle_slot is used only speculatively, hash computations are done on finality